test/Vesting.test.js

```diff
@@ -0,0 +1,288 @@
+const Vesting = artifacts.require("./Vesting");
+const MyToken = artifacts.require("./MyToken");
+
+const {
+  ether,           // Big Number support
+  constants,    // Common constants, like the zero address and largest integers
+  expectEvent,  // Assertions for emitted events
+  expectRevert,
+  balance,
+  time, // Assertions for transactions that should fail
+} = require('@openzeppelin/test-helpers');
+
+const { expect } = require("chai");
+const BN = Web3.utils.BN;
+
+contract("Vesting", async ([owner, acc2, acc3, acc4]) => {
+
+  let instanceToken;
+  let instanceVesting;
+
+  before(async () => {
+    instanceToken = await MyToken.new();
+    instanceVesting = await Vesting.new(instanceToken.address);
+  });
+
+  describe("Deploy Vesting contract - false", async () => {
+    it("Error : Incorrect address , only contract address", async () => {
+      await expectRevert(Vesting.new(acc2), "Error : Incorrect address , only contract address");
+    });
+  });
+
+  describe("Rigth initialization", async () => {
+    it("check owner balance - result 100000 tokens", async () => {
+      let balance = await instanceToken.balanceOf(owner);
+      expect(balance.toString()).to.be.equal(Web3.utils.toWei("100000", "ether"));
+    });
+  });
+
+  describe("setInitialTimestamp function", async () => {
+
+    describe("setInitialTimestamp function - false", async () => {
+      it("setInitialTimestamp - caller is not the owner", async () => {
+        let vestingStartDateBefore = await instanceVesting.vestingStartDate.call();
```

Comment:
.call() - this is superfluous, you can use call - instanceVesting.vestingStartDate() without .call